### PR TITLE
Fix universal binary for macOS

### PR DESCRIPTION
**Pull request by aviramha**

### CODE REVIEW

**Comment by aviramha** [comment]
@messense would love your feedback on it.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -12,13 +12,6 @@ jobs:
           - os: ubuntu-18.04
```
Comment (eyalb181) [inline .github/workflows/release.yaml:12]: Is there a reason to leave this as a matrix?

**Review by eyalb181 — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -39,6 +32,43 @@ jobs:
...
+          MIRRORD_LAYER_FILE: /tmp/target/universal-apple-darwin/release/libmirrord_layer.dylib
+        run: cargo +nightly build --release -p mirrord --target=aarch64-apple-darwin
+      - name: build macOS x86-64 cli with universal dylib
+        env:
+          MIRRORD_LAYER_FILE: /tmp/target/universal-apple-darwin/release/libmirrord_layer.dylib
+        run: cargo +nightly build --release -p mirrord --target=x86_64-apple-darwin
+      - name: create universal macOS cli
+        run: lipo -create -output /tmp/target/universal-apple-darwin/release/mirrord target/aarch64-apple-darwin/release/mirrord target/x86_64-apple-darwin/release/mirrord
+      - name: change dir to tmp
+        run: cd /tmp
```
Comment (messense) [inline .github/workflows/release.yaml:64]: I was under the impression that CWD is not preserved between steps?

File: mirrord-cli/build.rs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+fn main() {
+    if std::env::var("MIRRORD_LAYER_FILE").is_err() {
+        println!(
+            "cargo:rustc-env=MIRRORD_LAYER_FILE={}",
+            std::env::var("CARGO_CDYLIB_FILE_MIRRORD_LAYER").unwrap()
```
Comment (messense) [inline mirrord-cli/build.rs:5]: Nice trick! I wonder if there are any size limits for env vars.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (aviramha) [reply .github/workflows/release.yaml]: Good point, too bad upload artifacts is a very awful action.. I changed to copy the tmp target to current dir so it'll work now :)

File: .github/workflows/release.yaml (hunk 1 above)
Comment (aviramha) [reply .github/workflows/release.yaml:12]: erm, maybe when we want to have more archs

**Review by messense — APPROVED** [review]

**Pull request merged**